core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -425,6 +445,9 @@ UNWIND(	.cantunwind)
 
 	bl	plat_cpu_reset_late
 
+    /* this core just comes out of reset, discard undefined l1 contents */
+	bl	arm_cl1_d_invbysetway
```

Comment:
Could you cross check this? If core is inside the smp coherency, could this affect other cores.